kafka-client/src/main/java/dev/responsive/kafka/internal/clients/OffsetTracker.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * This source code is licensed under the Responsive Business Source License Agreement v1.0
+ * available at:
+ *
+ * https://www.responsive.dev/legal/responsive-bsl-10
+ *
+ * This software requires a valid Commercial License Key for production use. Trial and commercial
+ * licenses can be obtained at https://www.responsive.dev
+ */
+
+package dev.responsive.kafka.internal.clients;
+
+import java.util.BitSet;
+
+/**
+ * This class allows us to efficiently count the number of events
+ * between two offsets that match a certain condition. This is somewhat
+ * memory efficient in that we can track 100K offsets with ~1.5K longs
+ * (64 bits per long), or roughly 12KB.
```

Comment:
I'm a little confused by the point of this class. This is just counting the number of origin events since the last commit, right? It doesn't seem like we ever expose what the actual origin event offsets are, so why do we need to save them in this `BitSet` scheme -- ie why not just use an int that increments in `#mark` and resets/returns in `#countAndShift`?

Lmk if you'd rather sync online because I'm sure I'm missing something here 🙂

Reply:
This class should definitely have javadoc explaining it, originally I had implemented what you're saying but Rohan pointed out these two concerns: https://github.com/responsivedev/responsive-pub/pull/424#discussion_r1965094537 and https://github.com/responsivedev/responsive-pub/pull/424#discussion_r1965097847 both fixed by this bitset tracker.

Reply:
ah, I missed those discussions, but yeah he's right. thanks for filling in the javadocs